src/redux/actions/candidateActions.js

```diff
@@ -0,0 +1,25 @@
+import candidateApi from "../../apis/candidate";
+import { ActionTypes } from "../constants/action-types";
+
+export const setCandidates = (candidates) => {
+  return {
+    type: ActionTypes.SET_CANDIDATES,
+    payload: candidates,
+  };
+};
+
+export const sortedCandidates = (payload) => {
+  return {
+    type: ActionTypes.SORTED_CANDIDATES,
+    payload: payload,
+  };
+};
+
+export const fetchCandidates = () => async (dispatch) => {
+  const response = await candidateApi.get("/candidates");
+
+  dispatch({
+    type: ActionTypes.FETCH_CANDIDATES,
+    payload: response["data"]["data"]["data"],
```

Comment:
As discussed you can either destructure it or use response.data.data.data if not able to do then try JSON.parse(response) and then set in your payload.